src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -69,11 +70,11 @@ runtime::gpu::CUDAEmitter::CUDAEmitter(runtime::gpu::GPUPrimitiveEmitter* emitte
 
 size_t runtime::gpu::CUDAEmitter::build_pad(const runtime::gpu::GPURuntimeContext* ctx,
                                             const std::array<std::string, 2>& dtypes,
-                                            const Shape& input_shape,
-                                            const Shape& output_shape,
-                                            const Shape& padding_below,
-                                            const Shape& padding_above,
-                                            const Shape& padding_interior,
+                                            GPUShape input_shape,
```

Comment:
Why not pass these in by reference? I know this is during build time so it is not a big deal, but...

Reply:
Basically, I have to do a copy regardless to convert the Shape to the GPUShape, and so I figure the cleanest way to do that is to do so implicitly in the function call using the GPUShape conversion constructors. As a const reference, the compiler won't do the conversion and then the user would have to do it manually with (e.g.) `GPUShape(input_shape)` for every argument, for every function call.